source/jormungandr/jormungandr/new_relic.py

```
@@ -94,3 +95,33 @@ def ignore():
         except:
             logger = logging.getLogger(__name__)
             logger.exception('failure while ignoring transaction')
+
+
+class DistributedEvent:
+    """
+    Custom event that we publish to New Relic for distributed scenario
+    """
+
+    def __init__(self, service, call_name, group_name):
+        self.event_params = {
+            "service": type(service).__name__,
+            "call": call_name,
+            "group": group_name,
+            "status": "ok",
+        }
+
+    def time_function(self, function, *args, **kwargs):
+        start_time = timeit.default_timer()
+        result = None
+        try:
+            result = function(*args, **kwargs)
+        except Exception as e:
```

You probably want to reraise this exception